dclab/rtdc_dataset/feat_temp.py

```diff
@@ -70,12 +72,17 @@ def set_temporary_feature(rtdc_ds, feature, data):
     if not feat_logic.feature_exists(feature):
         raise ValueError(
             f"Temporary feature '{feature}' has not been registered!")
-    if isinstance(rtdc_ds, RTDC_Hierarchy):
-        raise NotImplementedError("Setting temporary features for hierarchy "
-                                  "children not implemented yet!")
     if len(data) != len(rtdc_ds):
         raise ValueError(f"The temporary feature {feature} must have same "
                          f"length as the dataset. Expected length "
                          f"{len(rtdc_ds)}, got length {len(data)}!")
+    if isinstance(rtdc_ds, RTDC_Hierarchy):
+        root_ids = map_indices_child2root(rtdc_ds, np.arange(len(rtdc_ds)))
+        root_parent = rtdc_ds.get_root_parent()
+        root_feat_data = np.empty((len(root_parent)))
+        root_feat_data[:] = np.nan
+        root_feat_data[root_ids] = data
+        feat_logic.check_feature_shape(feature, data)
+        root_parent._usertemp[feature] = root_feat_data
     feat_logic.check_feature_shape(feature, data)
```

Comment:
These lines should not be executed if we are an RTDC_Hierarchy, correct?. You can accomplish that with an `else` case.

Reply:
As I understand the code, we have prepare the temporary feature data (-> `root_feature_data`) that will be set in the root_parent. That has to be prepared using the indices of the hierarchy child, and then we can call `set_temporary_feature(root_parent, feature, root_feature_data)` as you suggested in the other comment. So we have to call it here when `rtdc_ds` is the hierarchy child while we still have access to the hierarchy child and its index information.

Reply:
Not sure we are talking about the same thing, but I think it should be enough to simply update the root parent and then call `rejuvenate` or so?

We don't actually need these two lines if we have a hierarchy child:
```
feat_logic.check_feature_shape(feature, data)
rtdc_ds._usertemp[feature] = data
```

Reply:
In any case, with or without an `else` statement, the lines of code will be executed once: Either when the data for the hierarchy child are being set or in the second function call of `set_temporary_feature` (from within the first function call), when the data for the root_parent are set.  

For me, it makes more sense to have it the way it is.

Reply:
> Not sure we are talking about the same thing, but I think it should be enough to simply update the root parent and then call `rejuvenate` or so?
> 
> We don't actually need these two lines if we have a hierarchy child:
> 
> ```python
> feat_logic.check_feature_shape(feature, data)
> rtdc_ds._usertemp[feature] = data
> ```

Is the child updated accordingly when the root_parent changes? And that would mean we only have to set the data once, acutally in the root_parent, and the child would automatically get the data? I am not sure if I fully understand the concept and ideas how you implemented this hierarchical structure of datasets...

Reply:
For me, it is not clear from the [docs](https://dclab.readthedocs.io/en/stable/sec_code_reference.html#dclab.rtdc_dataset.RTDC_Hierarchy) whether **adding data** to the parent also adds that data to the child. I think with regard to temporary features, this needs a bit of clarification.

Reply:
Yes, this is the case. For instance, `__contains__` always asks the parent first:

https://github.com/DC-analysis/dclab/blob/0.47.4/dclab/rtdc_dataset/fmt_hierarchy.py#L312-L313

same is the case for `__getitem__`:

https://github.com/DC-analysis/dclab/blob/0.47.4/dclab/rtdc_dataset/fmt_hierarchy.py#L315-L336

Reply:
aha, now I get it. Yes, than it makes sense what you say and I will change it asap.

Reply:
BTW this also works if you have multiple children. They would also get their share of the temporary feature when it's set for the root parent.